### PR TITLE
enforce c contiguous array, update napf, ++version[2]

### PR DESCRIPTION
**Pull request by j042**
# Overview
As cpp assumed c-contiguous memory layout, we need to make sure any arrays that goes to cpp side should fulfill this assumption. This PR does so and adds tests. Also fixes an error inside python approximate_curve.

## Addressed issues
*  correct spline init even with non-contiguous array, for example, transposed array
* warning fixes
* approximate curve fix - pop residual before spline init

## Checklists
* [x] Documentations are up-to-date.
* [ ] Added example(s)
* [x] Added test(s)

### CODE REVIEW

**Review by jzwar — APPROVED** [review]
LGTM! Thanks for saving my evening here. Can the line in microstructure-create now be deleted?

File: splinepy/bspline.py [hunk 1]
```diff
@@ -280,8 +284,10 @@ def approximate_curve(
                 centripetal=centripetal,
                 knot_vector=knot_vector,
         )
+
+        res = results.pop("residual")
+
         fitted = cls(**results)
-        res = results["residual"]
 
         utils.log.debug("BSpline curve approximation complete. ")
```
[the remark this answers is not in the document]
Comment (j042) [reply splinepy/bspline.py:292]: this is a classmethod, meaning that _logd is not available

[the remark this answers is not in the document]
Comment (j042) [reply splinepy/bspline.py:292]: I think I am wrong

File: splinepy/utils/data.py [hunk 2]
```diff
@@ -204,21 +204,60 @@ def is_modified(array):
...
     for key, value in dict_.items():
-        if value is not None:
-            dict_without_none[key] = value
+        # most likely will be asking about np.ndarray
+        if isinstance(value, np.ndarray):
+            value = enforce_contiguous(value)
+
+        # or, list. only check if the first element is np.ndarray
+        elif isinstance(value, list):
+            if isinstance(value[0], np.ndarray):
```
[the remark this answers is not in the document]
Comment (j042) [reply splinepy/utils/data.py:253]: as value[0] is instance, I'd like to keep this check

**Comment by j042** [comment]
yes, you can delete the line

**Pull request merged**